lib/ClangImporter/SwiftLookupTable.cpp

```diff
@@ -1606,7 +1606,8 @@ SwiftNameLookupExtension::hashExtension(llvm::hash_code code) const {
   return llvm::hash_combine(code, StringRef("swift.lookup"),
                             SWIFT_LOOKUP_TABLE_VERSION_MAJOR,
                             SWIFT_LOOKUP_TABLE_VERSION_MINOR,
-                            inferImportAsMember);
+                            inferImportAsMember,
+                            version::getSwiftFullVersion());
```

Comment:
I would have expected this to match what we add to `metadata.UserInfo` above - should this use `swiftCtx.LangOpts.EffectiveLanguageVersion` or maybe the user info should remove it?  Apropos, should there be a test that checks that output using `clang -module-file-info`?

Reply:
The effective language version already makes its way into the hash via API notes, but if you think it's simpler to just have them do the same thing I can change it. I don't see how I can write a test for the hash changing, though, since this one's about what happens if you change how the compiler is built.

Reply:
Ahh, now I see that setting the effective version just makes it print a *superset* of the version information, not *different* information.  Nevermind, it's fine as-is.

> I don't see how I can write a test for the hash changing

I agree, you can't really test the hashing.  I was thinking it would be nice to test what we're printing with `clang -module-file-info` for the extension, since the principle is that anything affecting the hash should get printed there. But that's not something you need to do for this patch, sorry for not making that clear.